network/network_test.go

```diff
@@ -104,7 +106,8 @@ func readData(t *testing.T, r io.ReadCloser, len int) []byte {
 }
 
 func TestStoppingNetwork(t *testing.T) {
-	net, err := NewNetwork(testConfig())
+	ctx := context.Background()
```

Comment:
for test you can use `context.ToDo`

Reply:
is that the just thing I should change?

Reply:
> is that the just thing I should change?

Yes

Reply:
ok, I will change it too
should I remove defer func for cancel() in node.go?

Reply:
> ok, I will change it too should I remove defer func for cancel() in node.go?

Yes